core/src/main/scala/org/apache/spark/util/collection/WritablePartitionedPairCollection.scala

```diff
@@ -74,6 +74,31 @@ private[spark] object WritablePartitionedPairCollection {
     }
   }
 
+  /* Takes an optional parameter (keyComparator), use if provided
+   * and returns a comparator for the partitions
+   */
+  def getComparator[K](keyComparator: Option[Comparator[K]]) : Comparator[(Int, K)] = {
+    val comparator : Comparator[(Int, K)] =
```

Comment:
`comparator` is now entirely redundant. The whole body is just the `if` statement